src/core/server/saved_objects/migrations/kibana/kibana_migrator.ts

```diff
@@ -95,8 +95,10 @@ export class KibanaMigrator {
    *    The promise resolves with an array of migration statuses, one for each
    *    elasticsearch index which was migrated.
    */
-  public runMigrations(): Promise<Array<{ status: string }>> {
-    if (this.migrationResult === undefined) {
+  public runMigrations({ rerun = false }: { rerun?: boolean } = {}): Promise<
+    Array<{ status: string }>
+  > {
+    if (this.migrationResult === undefined || rerun) {
       this.migrationResult = this.runMigrationsInternal();
     }
```

Comment:
This change is very naive: if a migration is already running and we call `runMigration` again with `return=true`, multiple migrations can occur at the same time. 

We should maybe wait for the pending migration to finish before triggering the next one. But that adds some complex logic to handle edge cases such as multiple `rerun=true` calls when a migration is already running. We should wait for the migration to finish but only re-execute ONCE and return the same promise for every calls to `rerun` that occurred during the initial migration.

As this API is internal I though current implementation was acceptable. Tell me if we should fireproof it more.

Reply:
so, in theory, it could lead to a race condition if 2 migrations run over the same data?

Reply:
In theory it could yes, even if I don't think it could leads to data corruption, as in worse case, both 'threads'  will access the data before any actually migrated it, and the document would be migrated then updated 'twice' from the same original data, which would only impact performances.

I can try to improve that by 'pooling' all the calls to `runMigration(rerun=true)` while one migration is running and only execute the rerun once, as explained in previous comment, if we think current implementation is too fragile.

Reply:
It's okay as long as this API is used only for tests internally. Could you add a comment about this known limitation?